state/unit.go

```diff
@@ -739,12 +734,12 @@ func (op *DestroyUnitOperation) destroyOps() ([]txn.Op, error) {
 		Id:     op.unit.st.docID(agentStatusDocId),
 		Assert: bson.D{{"status", agentStatusInfo.Status}},
 	}
-	removeAsserts := append(isAliveDoc, bson.DocElem{
+	removeAsserts := bson.D{{
```

Comment:
What we typically do elsewhere is assert the life doesn't change (usually only if force==true).
This covers the case where we want to re-evaluate things if the unit goes alive->dying